sdk/messaging/azservicebus/internal/atom/entity_manager.go

```diff
@@ -238,61 +243,90 @@ func (em *entityManager) Delete(ctx context.Context, entityPath string, mw ...Mi
 }
 
 func (em *entityManager) execute(ctx context.Context, method string, entityPath string, body io.Reader, mw ...MiddlewareFunc) (*http.Response, error) {
-	ctx, span := em.startSpanFromContext(ctx, "sb.ATOM.Execute")
-	defer span.End()
+	var finalResp *http.Response
 
-	req, err := http.NewRequest(method, em.Host+strings.TrimPrefix(entityPath, "/"), body)
-	if err != nil {
-		tab.For(ctx).Error(err)
-		return nil, err
-	}
+	err := utils.Retry(ctx, fmt.Sprintf("%s %s", method, entityPath), func(ctx context.Context, args *utils.RetryFnArgs) error {
+		ctx, span := em.startSpanFromContext(ctx, "sb.ATOM.Execute")
+		defer span.End()
 
-	final := func(_ RestHandler) RestHandler {
-		return func(reqCtx context.Context, request *http.Request) (*http.Response, error) {
-			client := &http.Client{
-				Timeout: 60 * time.Second,
-			}
-			request = request.WithContext(reqCtx)
-			return client.Do(request)
+		req, err := http.NewRequest(method, em.Host+strings.TrimPrefix(entityPath, "/"), body)
+		if err != nil {
+			tab.For(ctx).Error(err)
+			return err
 		}
-	}
 
-	mwStack := []MiddlewareFunc{final}
-	sl := len(em.mwStack) - 1
-	for i := sl; i >= 0; i-- {
-		mwStack = append(mwStack, em.mwStack[i])
-	}
+		final := func(_ RestHandler) RestHandler {
+			return func(reqCtx context.Context, request *http.Request) (*http.Response, error) {
+				client := &http.Client{
+					Timeout: 60 * time.Second,
+				}
+				request = request.WithContext(reqCtx)
+				return client.Do(request)
+			}
+		}
 
-	for i := len(mw) - 1; i >= 0; i-- {
-		mwStack = append(mwStack, mw[i])
-	}
+		mwStack := []MiddlewareFunc{final}
+		sl := len(em.mwStack) - 1
+		for i := sl; i >= 0; i-- {
+			mwStack = append(mwStack, em.mwStack[i])
+		}
 
-	var h RestHandler
-	for _, mw := range mwStack {
-		h = mw(h)
-	}
+		for i := len(mw) - 1; i >= 0; i-- {
+			mwStack = append(mwStack, mw[i])
+		}
 
-	resp, err := h(ctx, req)
+		var h RestHandler
+		for _, mw := range mwStack {
+			h = mw(h)
+		}
 
-	if err == nil {
-		if resp.StatusCode >= http.StatusBadRequest {
-			bytes, err := ioutil.ReadAll(resp.Body)
+		resp, err := h(ctx, req)
 
-			if err == nil {
-				err = FormatManagementError(bytes, err)
+		if err == nil {
+			if resp.StatusCode >= http.StatusBadRequest {
+				return NewResponseError(resp)
 			}
 
-			return nil, NewResponseError(err, resp)
+			finalResp = resp
+			return nil
+		}
+
+		if resp != nil {
+			return NewResponseError(resp)
 		}
 
-		return resp, nil
+		return err
+	}, isFatalHTTPError, em.retryOptions)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return finalResp, nil
+}
+
+func isFatalHTTPError(err error) bool {
+	var netErr net.Error
+
+	if errors.As(err, &netErr) {
```

Comment:
I think this can go away.

Reply:
For better or worse my default in this function is to assume an error is fatal. I think we talked about making it so network errors are just always treated as retryable (so that's all this is doing).